src/main/java/io/reactivex/internal/functions/Functions.java

```diff
@@ -240,4 +243,367 @@ public int compare(Object a, Object b) {
+     * @return the new Action instance
+     */
+    public static Action futureAction(Future<?> future) {
+        return new FutureAction(future);
+    }
+    
+    static final class JustValue<T, U> implements Callable<U>, Function<T, U> {
+        final U value;
+        
+        public JustValue(U value) {
+            this.value = value;
+        }
+        
+        @Override
+        public U call() throws Exception {
+            return value;
+        }
+        
+        @Override
+        public U apply(T t) throws Exception {
+            return value;
+        }
+    }
+    
+    /**
+     * Returns a Callable that returns the given value.
+     * @param <T> the value type
+     * @param value the value to return
+     * @return the new Callable instance
+     */
+    public static <T> Callable<T> justCallable(T value) {
+        return new JustValue<Object, T>(value);
+    }
+
+    /**
+     * Returns a Function that ignores its parameter and returns the given value.
+     * @param <T> the function's input type
+     * @param <U> the value and return type of the function
+     * @param value the value to return
+     * @return the new Function instance
+     */
+    public static <T, U> Function<T, U> justFunction(U value) {
+        return new JustValue<T, U>(value);
+    }
+
+    static final class CastToClass<T, U> implements Function<T, U> {
+        final Class<U> clazz;
+        
+        public CastToClass(Class<U> clazz) {
+            this.clazz = clazz;
+        }
+        
+        @Override
+        public U apply(T t) throws Exception {
+            return clazz.cast(t);
+        }
+    }
+    
+    /**
+     * Returns a function that cast the incoming values via a Class object.
+     * @param <T> the input value type
+     * @param <U> the output and target type
+     * @param target the target class
+     * @return the new Function instance
+     */
+    public static <T, U> Function<T, U> castFunction(Class<U> target) {
+        return new CastToClass<T, U>(target);
+    }
+
+    static final class ArrayListCapacityCallable<T> implements Callable<List<T>> {
+        final int capacity;
+        
+        public ArrayListCapacityCallable(int capacity) {
+            this.capacity = capacity;
+        }
+        
+        @Override
+        public List<T> call() throws Exception {
+            return new ArrayList<T>(capacity);
+        }
+    }
+    
+    public static <T> Callable<List<T>> createArrayList(int capacity) {
+        return new ArrayListCapacityCallable<T>(capacity);
+    }
+    
+    static final class EqualsPredicate<T> implements Predicate<T> {
+        final T value;
+        
+        public EqualsPredicate(T value) {
+            this.value = value;
+        }
+        
+        @Override
+        public boolean test(T t) throws Exception {
+            return Objects.equals(t, value);
+        }
+    }
+    
+    public static <T> Predicate<T> equalsWith(T value) {
```

Comment:
Is there any reason to use these static factories? The nested classes could just be public (or promoted to top-level) and reference directly.